### ISSUE TITLE
xerces: compiler warning

### ISSUE THREAD

**Issue by markus2330** [ElektraInitiative/libelektra#1454]
Seems to be a useful warning:

```
src/plugins/xerces/serializer.cpp: In function ‘void xerces::serialize(const kdb::Key&, const kdb::KeySet&)’:
src/plugins/xerces/serializer.cpp:63:80: warning: ‘child’ may be used uninitialized in this function [-Wmaybe-uninitialized]
    elem.setAttribute (asXMLCh (meta.getName ()), asXMLCh (meta.get<string> ()));
                                                                                ^
src/plugins/xerces/serializer.cpp:119:15: note: ‘child’ was declared here
  DOMElement * child;
               ^
```

**Comment by e1528532** [comment]
Checked the code out and i think its a false positive. Keep in mind that the usage happens in key2xml and look at the code where child is initialized:

```
        if (name == key.end ()) throw new XercesPluginException ("Key " + key.getFullName () + " is not under " + parentKey.getFullName ());
        // ... stuff
	DOMElement * child;
	for (; name != key.end (); name++) {
                // ... stuff
		child = arrayChild ? arrayChild : findChildWithName (*current, actualName);

		if (!child) {
			ELEKTRA_LOG_DEBUG ("creating path element %s", actualName.c_str ());
			child = doc.createElement (asXMLCh (actualName));
			current->appendChild (child);
		}
		current = child;
	}

	key2xml (doc, *child, actualName, key);
```

As you can see, due to the exception above we guarantee that name is not at key.end () at the beginning of the loop, thus it will at least go inside the loop once, thus it will create a new document element in all cases so it won't be uninitialized. According to https://xerces.apache.org/xerces-c/apiDocs-3/classDOMDocument.html#af5e93a6b757adb2544b3f6dffb4b461a it seems that createElement either returns an element or throws an exception, so, unless i have overseen something i think its a false positive. My compiler clang (Apple LLVM version 8.1.0 (clang-802.0.38)) doesn't report anything either when using -Weverything 

However, i found a minor issue (using new with throw in c++) and looks like one of the testrecorder fixes broke the xerces shell test, i'll take a look at those two tomorrow

**Comment by markus2330** [comment]
Thank you for looking into it! Please fix the warning nevertheless, you can suppress it and put your above comment as rationale there.

**Comment by e1528532** [comment]
fixed in #1508
